lib/src/git.rs

```diff
@@ -1143,10 +1144,43 @@ pub fn rename_remote(
     Ok(())
 }
 
+fn looks_like_a_path_to_git(source: &str) -> bool {
+    source.chars().nth(1) == Some(':') || // Looks like a Windows C:... path
+    // Match Git's behavior, the URL syntax [is only recognized if there are no
+    // slashes before the first
+    // colon](https://git-scm.com/docs/git-clone#_git_urls)
+    !source
+        .split_once("/")  
+        .map_or(source, |(first, _)| first)
+        .contains(":")
+}
+
+pub fn sanitize_git_url_if_path(cwd: &Path, source: &str) -> String {
+    // Git appears to turn URL-like source to absolute path if local git directory
+    // exits, and fails because '$PWD/https' is unsupported protocol. Since it would
+    // be tedious to copy the exact git (or libgit2) behavior, we simply assume a
+    // source containing ':' is a URL, SSH remote, or absolute path with Windows
+    // drive letter.
+    if looks_like_a_path_to_git(source) && Path::new(source).exists() {
+        // TODO: This won't work for Windows UNC path or (less importantly) if the
+        // original source is a non-UTF Windows path. For Windows UNC paths, we
+        // could use dunce, though see also https://gitlab.com/kornelski/dunce/-/issues/7
+        // TODO: double-check about UNC paths; what does Git do?
+        cwd.join(source).to_slash().map_or_else(
```

Comment:
Can't we simply do replace `MAIN_SEPARATOR` to slash (if `cfg!(windows)`)? `path-slash` might be optimized if we have to deal with `Path`/`OsStr`, but here we build `String` anyway.

Reply:
This is on my list to try and see if it makes things saner. My main worry is about UNC paths, especially the ones appearing in our tests. I would have to find out experimentally whether I can get rid of them via `dunce`, and/or I'd have to find out how libgit2 deals with paths like:

- `\\?\C:/Path/`
- `//?/C:/Path/`

Neither of these is correct according to Microsoft, but I think `libgit2` definitely chokes on the correct `\\?\C:\Path\`. I wish I knew what Git did; I think we'd have to ask or read the source to find out.

(As for dependency on path-slash, that library hasn't been updated in years and it's probably pretty simple, so we could re-implement it if needed)